### PR TITLE
Add health endpoints

### PR DESCRIPTION
**Pull request by Sotatek-HuyLe3a**
#49

### CODE REVIEW

File: application/src/main/java/org/cardanofoundation/ledgersync/listeners/BlockEventListener.java [hunk 1]
```diff
@@ -109,6 +121,10 @@ public void handleRollback(RollbackEvent rollbackEvent) {
         rollbackService.rollBackFrom(rollbackBlockNo);
         metricCollectorService.collectRollbackMetric();
         blockCount.set(0);
+
+        healthCheckCachingService.saveLatestBlockSlot(rollbackEvent.getRollbackTo().getSlot());
+        healthCheckCachingService.saveLatestBlockInsertTime(LocalDateTime.now(ZoneOffset.UTC));
+        healthCheckCachingService.saveLatestBlockTime(rollBackBlock.get().getTime().toLocalDateTime());
```
Comment (satran004) [inline application/src/main/java/org/cardanofoundation/ledgersync/listeners/BlockEventListener.java:127]: Do we need to use Zone (UTC) like it's done for genesis block above?

Or, should we just keep latestBlockTime as long?  AFAIK, blockTime is actually in seconds. So, we can get the current time in seconds and find the difference.

File: application/src/main/java/org/cardanofoundation/ledgersync/dto/healthcheck/HealthStatus.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+import java.time.LocalDateTime;
+
+@Getter
+@Setter
+@AllArgsConstructor
+@NoArgsConstructor
+@Builder
+public class HealthStatus {
+    Boolean isHealthy;
+    String message;
```
Comment (satran004) [inline application/src/main/java/org/cardanofoundation/ledgersync/dto/healthcheck/HealthStatus.java:14]: Should we keep another field for message code?

For example:
message_code = READY_TO_SERVE
messge_desc = "Data is ready to serve"

File: application/src/main/resources/config/application.yaml [hunk 3]
```diff
@@ -85,4 +85,22 @@ ledger-sync:
     pool-offline-data:
       fixed-delay: ${POOL_OFFLINE_DATA_FIXED_DELAY:172800}
       initial-delay: ${POOL_OFFLINE_DATA_INIT_DELAY:20000}
-    enabled: ${SCHEDULER_ENABLED:true}
\ No newline at end of file
+    enabled: ${SCHEDULER_ENABLED:true}
+  healthcheck:
+    block-time-threshold: ${BLOCK_TIME_THRESHOLD_IN_SECOND:240}
+    inserted-time-threshold: ${INSERTED_TIME_THRESHOLD_IN_SECOND:1200}
```
Comment (satran004) [inline application/src/main/resources/config/application.yaml:91]: Isn't that too long? 20 minutes

**Review by satran004 — COMMENTED** [review]
Thanks @Sotatek-HuyLe3a. Few comments.

File: application/src/main/resources/config/application.yaml (hunk 3 above)
Comment (Sotatek-HuyLe3a) [reply application/src/main/resources/config/application.yaml:91]: This threshold is used for both cases: batching and non-batching, so I set the default value to be larger than normal.
We could use two separate thresholds for each case. However, the drawback is that we have to add environment variables, which could introduce unnecessary complexity.

File: application/src/main/java/org/cardanofoundation/ledgersync/listeners/BlockEventListener.java (hunk 1 above)
Comment (Sotatek-HuyLe3a) [reply application/src/main/java/org/cardanofoundation/ledgersync/listeners/BlockEventListener.java:127]: In this case, toLocalDateTime returns the correct value, rollBackBlock.get().getTime() returns the Timestamp value that was previously stored with UTC+0
We could keep latestBlockTime as long, but in rollback case as above, we cannot save a correct "long" value for latest block time

**Comment by Sotatek-HuyLe3a** [comment]
@satran004, I updated, please review again. Thanks.

**Review by satran004 — APPROVED** [review]

**Pull request merged**